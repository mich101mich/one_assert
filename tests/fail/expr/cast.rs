fn main() {
    one_assert::assert!(1 as bool);
    one_assert::assert!(1 as i32);
}
