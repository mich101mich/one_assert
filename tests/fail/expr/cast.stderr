error[E0308]: mismatched types
 --> tests/fail/expr/cast.rs:3:25
  |
3 |     one_assert::assert!(1 as i32);
  |                         ^^^^^^^^ expected `bool`, found `i32`

error[E0054]: cannot cast `{integer}` as `bool`
 --> tests/fail/expr/cast.rs:2:25
  |
2 |     one_assert::assert!(1 as bool);
  |                         ^^^^^^^^^
  |
help: compare with zero instead
  |
2 |     one_assert::assert!(1 != 0);
  |                           ~~~~
